hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -858,7 +821,56 @@ && isMemberRejoining(memberMap, member.getAddress(), member.getUuid())) {
         }
     }
 
-    private void runProvidedPostJoinOpsWithoutBroadcast() {
+    private boolean shouldTriggerRepartitionSyncStrategyOnly(MemberMap memberMap) {
+        boolean trigger = true;
+        if (syncJoinStrategy != null) {
+            for (BiTuple<MemberInfo, OnJoinOp> tuple : joiningMembers.values()) {
+                MemberInfo member = tuple.element1();
+                if (isMemberRestartingWithPersistence(member.getAttributes())
+                    && isMemberRejoining(memberMap, member.getAddress(), member.getUuid())) {
+                    logger.info(member + " is rejoining the cluster");
+                    // do not trigger repartition immediately, wait for joining member to load hot-restart data
+                    trigger = false;
+                }
+            }
+        }
+        return trigger;
```

Comment:
Unless we explicitly want to print logger output for every rejoining member, we don't need a separate `trigger` variable here, we can just `return false;` within our loop and `return true;` at the end.

Reply:
ok sounds good